Failing test with bundled BOOST

On my system, using the bundled BOOST version, the following tests fail:
```
        3533 - dofs/nodal_renumbering_01.mpirun=2.debug (Failed)
        3535 - dofs/nodal_renumbering_01.mpirun=3.debug (Failed)
        3537 - dofs/nodal_renumbering_01.debug (Failed)
```
That's because of this failure:
```
3539: /home/fac/g/bangerth/p/deal.II/1/dealii/tests/dofs/nodal_renumbering_01.cc:38:10: fatal error: boost/algorithm/apply_permutation.hpp: No such file or directory
3539:    38 | #include <boost/algorithm/apply_permutation.hpp>
3539:       |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
3539: compilation terminated.
```

The code was added as part of #13536 by @drwells . We should either add that file to the bundled version, or find a different way to express the operation that doesn't require the boost algorithm.

--- a/tests/dofs/nodal_renumbering_01.cc
+++ b/tests/dofs/nodal_renumbering_01.cc
@@ -35,8 +35,6 @@
 #include <deal.II/numerics/data_out.h>
 #include <deal.II/numerics/vector_tools_interpolate.h>
 
-#include <boost/algorithm/apply_permutation.hpp>
-
 #include <iostream>
 #include <map>
 

